aspnetcore-features/HowToPreventAHangfireJobFromRunning/HowToPreventAHangfireJobFromRunning/Services/JobService.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace HowToPreventAHangfireJobFromRunning.Services;
+
+public class JobService
+{
+    private const int OperationDurationInSeconds = 10 * 60;
+
+    private readonly ILogger<JobService> _logger;
+
+    public JobService(ILogger<JobService> logger)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    public async Task RunJob1Async() => await PerformLongRunningOperationAsync(nameof(RunJob1Async));
```

Comment:
1. Add DisableConcurrentExecution attribute for RunJob1Async
2. Also add test case for this job as well

Reply:
While I can add this attribute, it is not working (at least while we're using SqlLite); a test was added, proving that this attribute is not working as expected. BTW it is not only my experience: https://github.com/HangfireIO/Hangfire/issues/1960

Reply:
One more thing: there is no way to test [DisableConcurrentExecution] with the in-memory storage. Everything works if we're using Microsoft SQL Server, but not for the InMemoryStorage nor the SqlLite provider:

    //config.UseInMemoryStorage();
    config.UseSqlServerStorage(builder.Configuration.GetConnectionString("HangfireConnection"),
        new SqlServerStorageOptions
        {
            PrepareSchemaIfNecessary = true,
        });
So, while I can test that this attribute is not working for InMemoryStorage, there is no easy way to test a working scenario with Microsoft SQL Server, which is the main reason why I skipped this test.